src/mame/igs/igs017_igs031.cpp

```diff
@@ -570,9 +570,9 @@ void igs017_igs031_device::tjsb_decrypt_sprites()
 	}
 }
 
-void igs017_igs031_device::tarzan_decrypt_sprites(size_t max_size)
+void igs017_igs031_device::tarzan_decrypt_sprites(size_t max_size, size_t flip_size)
 {
-	mgcs_flip_sprites(max_size);
+	mgcs_flip_sprites(flip_size);
 
 	const int rom_size = max_size ? max_size : memregion("sprites")->bytes();
 	u8 *rom = memregion("sprites")->base();
```

Comment:
This should throw a fatal error if `max_size` is greater than the size of the ROM region.  With the code as-is, it will read and then write off the end of the ROM region if someone inadvertently passes a `max_size` that’s too large.

Reply:
again you're nitpicking over me following the style of the logic already in the code.

Reply:
You’re trying to use the justification that because code already in MAME has a particular issue, it’s fine to add more code with the same issue.  That’s how issues propagate and technical debt builds up.

Reply:
and when people don't stick to the style already in MAME you mean that they're not doing that.

anyway, as I said, I've abandoned this now.  I was going above and beyond by even trying to get it to do something, when really the only interest I have in this is making sure the dumps we were getting were good.

my effort now drops to absolute minimum for this.